In [20]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

diamond_df = pd.read_csv("https://raw.githubusercontent.com/ktxdev/AIM-5001/main/M8/1.%20Data/diamonds.csv")

# Part 1: Data Summary

#### How many use cases?

In [21]:
len(diamond_df)

53940

#### How many attributes are in each use case

In [22]:
len(diamond_df.columns)

11

#### What the data types are for each of the attributes?

In [23]:
diamond_df.dtypes

Unnamed: 0      int64
carat         float64
cut            object
color          object
clarity        object
depth         float64
table         float64
price           int64
x             float64
y             float64
z             float64
dtype: object

# Part 2: Exploratory Data Analysis

## Cut
### Summary Statistics

In [24]:
diamond_df['cut'].describe()

count     53940
unique        5
top       Ideal
freq      21551
Name: cut, dtype: object

The cut attribute is composed of 5 unique categorical values, with the most frequently occurring value being 'Ideal', which has a frequency count of 21551.

### Possible values for the "Cut" attribute

In [25]:
diamond_df['cut'].unique()

array(['Ideal', 'Premium', 'Good', 'Very Good', 'Fair'], dtype=object)

### Value counts for the "Cut" attribute

In [26]:
diamond_df['cut'].value_counts()

cut
Ideal        21551
Premium      13791
Very Good    12082
Good          4906
Fair          1610
Name: count, dtype: int64